File: sdks/php/src/Models/ExtractResult.php
<?php

declare(strict_types=1);

namespace Renamed\Models;

/**
 * Result of extract operation.
 */
final class ExtractResult
{
    /**
     * @param array<string, mixed> $data Extracted data matching the schema.
     */
    public function __construct(
        /** Extracted data matching the schema. */
        public readonly array $data,
        /** Confidence score (0-1). */
        public readonly float $confidence,
    ) {}

    /**
     * Create instance from API response array.
     *
     * @param array<string, mixed> $data
     */
    public static function fromArray(array $data): self
    {
        return new self(
            data: $data['data'] ?? [],
            confidence: (float) ($data['confidence'] ?? 0.0),
        );
    }

    /**
     * Convert to array.
     *
     * @return array<string, mixed>
     */
    public function toArray(): array
    {
        return [
            'data' => $this->data,
            'confidence' => $this->confidence,
        ];
    }
}

File: sdks/csharp/Renamed/Models/ExtractResult.cs
using System.Text.Json.Serialization;

namespace Renamed.Sdk.Models;

/// <summary>
/// Result of extract operation containing structured data from the document.
/// </summary>
public sealed class ExtractResult
{
    /// <summary>
    /// Extracted data matching the provided schema.
    /// </summary>
    [JsonPropertyName("data")]
    public required IDictionary<string, object?> Data { get; init; }

    /// <summary>
    /// Confidence score (0-1) of the extraction.
    /// </summary>
    [JsonPropertyName("confidence")]
    public double Confidence { get; init; }
}
